### PR TITLE
Notify users in realtime for a new post

### PR DESCRIPTION
**Pull request by Devorein**
[Task Card](https://app.charmverse.io/charmverse/page-49366552253645923?viewId=d15c9b90-8918-44da-bbd8-db8712faa723&cardId=fffaec65-404b-4848-9395-4a1ecce7cdcb)

image

### CODE REVIEW

File: components/forum/components/ForumPosts/ForumPosts.tsx [hunk 1]
```diff
@@ -134,7 +182,9 @@ export default function ForumPosts({ search, categoryId }: ForumPostsProps) {
 
   return (
     <>
-      <CreateForumPost />
+      <Box ref={createPostBoxRef}>
```
Comment (mattcasey) [inline components/forum/components/ForumPosts/ForumPosts.tsx:185]: you could just use 'div' here

**Review by mattcasey — CHANGES_REQUESTED** [review]
needs to not use snackbar

File: hooks/useSnackbar.tsx [hunk 2]
```diff
@@ -43,11 +47,14 @@ export function SnackbarProvider({ children }: { children: ReactNode }) {
     if (reason === 'clickaway') {
       return;
     }
-
+    setOrigin({
```
Comment (mattcasey) [inline hooks/useSnackbar.tsx:50]: what happens when snackbar is used again? won't the origin be incorrect?
I think this feature is better off not using snackbar.. the utility is intended for quick messages in the bottom-left, which we use in a lot of places. whereas the 'new posts' option is very specific

**Review by mattcasey — CHANGES_REQUESTED** [review]
I think this feature is better off not using the global snackbar.. the utility is intended for quick messages in the bottom-left, which we use in a lot of places. whereas the 'new posts' option is a very specific UI component. (once you start adding lots of configuration props, overrides, etc. to a component it's a sign that it may be better to just build a custom one)

Also... looks like we were not creating posts originally until this PR? That needs to be fixed, and we need a "Publish" button too

File: lib/websockets/interfaces.ts [hunk 3]
```diff
@@ -60,6 +60,11 @@ type VotesUpdated = {
   payload: ExtendedVote[];
 };
 
+type PostCreated = {
+  type: 'post_created';
+  payload: Post & { page: Page | null };
```
Comment (mattcasey) [inline lib/websockets/interfaces.ts:65]: i don't think this event needs any payload, definitely not all the page content, etc.

File: pages/api/pages/index.ts [hunk 4]
```diff
@@ -121,6 +121,32 @@ async function createPageHandler(req: NextApiRequest, res: NextApiResponse<IPage
...
+            connect: {
+              id: newPageToNotify.id
+            }
+          },
+          status: 'draft'
+        }
+      });
+
+      if (post) {
+        relay.broadcast(
```
Comment (mattcasey) [inline pages/api/pages/index.ts:137]: this event should only be triggered when a post is published

File: hooks/useSnackbar.tsx (hunk 2 above)
Comment (Devorein) [reply hooks/useSnackbar.tsx:50]: Not if its the origin is updated beforehand. But I agree this is not the safest approach.

**Review by mattcasey — APPROVED** [review]

**Pull request merged**